rclcpp/include/rclcpp/experimental/serialized_container.hpp

```diff
@@ -0,0 +1,75 @@
+// Copyright 2019 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef RCLCPP__EXPERIMENTAL__SERIALIZED_CONTAINER_HPP_
+#define RCLCPP__EXPERIMENTAL__SERIALIZED_CONTAINER_HPP_
+
+#include <cstring>
+
+#include "rcutils/logging_macros.h"
+
+#include "rmw/serialized_message.h"
+
+namespace rclcpp
+{
+namespace experimental
+{
+
+/// Object oriented version of rcl_serialized_message_t with destructor to avoid memory leaks
+class SerializedContainer : public rcl_serialized_message_t
```

Comment:
The inheritance model goes okay here, yet I would prefer to be consistent with what we have in `rclcpp::LoanedMessage` and have the C++ class only wrap the `rcl_serialized_message_t`.